### DISCUSSION TITLE
A shorter CLI command

### DISCUSSION THREAD

**Discussion by zkochan** [pnpm/pnpm#3383] [Ideas]
I remember a few posts on Twitter about pnpm being hard to say and hard to type. I don't think we need to rename the project but we can create a shorter version of the CLI command. 

What if we will export an alias called `pn`?

So users will be able to use either `pnpm install` or `pn install`

**Comment by thynson** [comment]
I think the name shall be kept as is, but more shortcuts for sub-commands and options would be appreciated.

**Reply by zkochan** [reply]
Subcommands have short aliases already: i, rm, up, ls...

**Reply by thynson** [reply]
Yeah, even not frequently used subcommand has a shortcut. 
While  pnpm has 4 characters, which is at least the same with yarn, so I don't think it needs to be shorter.

**Reply by zkochan** [reply]
You can think about it as a feature. Less typing than npm or yarn. But as I said, it would be just an alias. pnpm would be still available.

**Reply by italodeandra** [reply]
@thynson I agree that 4 characters is already short, but I'm having a really hard time trying not to write `npnpm` for some unknown reason to me. This is totally a brain confusion but an alias like `pn` would be awesome.

**Reply by ibezkrovnyi** [reply]
4 letters is already short enough, but it's hard for me and for most devs in our teams to type `pnpm`. I mostly type `npnpm` or `pnm`, sometimes `npm`. idk why. Experimented a bit and if first letter has to be `p` (top rightmost letter on kb), then next letter has to be for the left hand, best - any leftmost letter, e.g. `pq`, `pa`, `pz`. After that right hand is already ready to type `--` to add some cli parameter

**Comment by sunnylqm** [comment]
Another big problem is, "pnpm" is not so good for search engine. It easily gets overshadowed.

![image](https://user-images.githubusercontent.com/615282/142114961-e4eff371-d2e5-4999-b67b-e2e21f375c19.png)

**Reply by zkochan** [reply]
yes, this was mentioned a lot on Twitter. This is a legitimate issue. Should we rename the project?

**Reply by sunnylqm** [reply]
I'll vote yes. It deserves a name that not related to npm at all.

**Reply by Danny-Rodriguez** [reply]
This might be an old thread but I’d say an alias would be appreciated. I used pnpm for the first time the other day from a tutorial and I had so much trouble consistently typing “pnpm”. It must be a cognitive process judging from others’ responses here.

**Reply by zkochan** [reply]
We have instructions how to make a shorter alias: https://pnpm.io/installation#using-a-shorter-alias

**Reply by IanGraingerGMSL** [reply]
> We have instructions how to make a shorter alias: https://pnpm.io/installation#using-a-shorter-alias

I can't see them? That anchor doesn't exist for me!

**Comment by Jack-Works** [comment]
I have a shell function that called `i`, which calls `npm i` `yarn` or `pnpm i` based on the lockfile in the current dir.

**Reply by zkochan** [reply]
There's also a tool that can make it automatically: https://github.com/antfu/ni

**Comment by kylemh** [comment]
Drop the “n”.

Just: “p”.

It’s cleaner.

![](https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQCvm715cEVVFcPWj0LAmWdzJnSQoAz6OdhJQ&usqp=CAU)

**Reply by zkochan** [reply]
That sounds like pee

**Comment by benschac** [comment]
Giving my two cents as a developer with dyslexia. The command `pnpm` is difficult to write. I'm constantly having to double or triple-check every time I write the command. Even just saying the name of the tool is kind of tricky.


While I could create an alias. I don't have to create an alias for `yarn` or `npm`. I'd love to not have to treat `pnpm` differently than its peers and take additional steps to use the tool.

I know I'm using free open source software and the solution is either to fix this myself or use a different tool.

What I'm asking is to bring the barrier of entry down. It's harder to get started with `pnpm` than its competitors.

**Reply by zkochan** [reply]
Would the "pn" alias be easier to write for you? I agree that the project name is hard to pronounce. I don't think a rename at this stage is a good idea. But we can change the way it is pronounced. Like I know some pronounce it as "pinpom"

**Reply by hiepxanh** [reply]
Yes I think an alias is just enough, we dont need to rename

**Reply by brianjlacy** [reply]
could alias it as pn and pronounce it "pun" with the tagline

"pn is more pun"

Edit: or simply "pn is pun"

**Comment by hiepxanh** [comment]
hello, any update on this? I love `pn` shortcut, add it please sir
 @zkochan

**Reply by zkochan** [reply]
This issue doesn't have a lot of upvotes. For now, you can create an alias yourself by putting this into your shell config:

```
alias pn=pnpm
```

**Reply by zkochan** [reply]
We may put a recipe about it to the installation page: https://pnpm.io/installation

**Reply by hiepxanh** [reply]
Yes, i think that is good place for everybody

**Reply by roll** [reply]
> This issue doesn't have a lot of upvotes. For now, you can create an alias yourself by putting this into your shell config:

Hi, now this is the second most popular all-time among open ones. Any chance you might reconsider?

**Comment by angelhdzmultimedia** [comment]
Thanks for the solution, aliases. I use Powershell 7 Core in Windows so in my case it was `set-alias -name pn -value pnpm`.

**Reply by zkochan** [reply]
Could you instructions how to do it to our docs?

See the `Using a shorter alias` section here https://pnpm.io/installation#using-a-shorter-alias

**Reply by angelhdzmultimedia** [reply]
If you are asking me to make a pull request of the documentation, sure, I will be more than pleased to contribute. Indeed, I was about to make a video yesterday for my YouTube tutorials channel, but couldn't record it due to little time. But I am going to record it today and then make the pull request.

**Reply by zkochan** [reply]
ok, if it is related, you can also add a link to your video.

**Reply by angelhdzmultimedia** [reply]
I made the pull request, and it's ready to be merged. I will make the video later because I didn't have the time.

**Comment by see2ever** [comment]
```bash
alias pn="pnpm"
alias pnr="pnpm run"
alias pns="pnpm start"
```

**Comment by brianjlacy** [comment]
Couple more arguments for a shorter "official" alias:

1. **It's surprisingly awkward.**

  - For a command typed as often as a package manager often is, it isn't that `pnpm` is "long" at 4 chars. It's that it's awkward to type (at least in my own experience) vs npm or yarn.

  - It's very easy to stumble over the letters and end up with `pnpn`, `pmpm`, `npmp` or some such thing (each of those examples, btw, are ones I've specifically made in the past, and am more likely to make in high pressure scenarios 😅).

2. **Chance of alias conflicts**
- It's true that aliases are easy enough to create on our own, and this is what I typically do. But without an *official* alias, the chances increase over time of a conflict between a high profile tool like `pnpm` and some other important tool that decides `pn` or `pmp` sounds like a keen CLI tool name, forcing me to change my alias (or create one for the other tool instead).

- If an official alias were added, it'd likely become the defacto usage for the tool, and other libraries would avoid using it (just as no one serious is going to name another CLI pnpm or yarn). As popular tools (and aliases) take root in the community, it effectively "reserves" them for use in the _defacto_ community name space so that other tool creators steer clear.

-----

**TL;DR** -- Throwing my support behind an OFFICIAL alias to
- (a) prevent keyboard stumbles
- (b) minimize chance of tool name/alias collisions over time

**Reply by roll** [reply]
I think another important point is that a custom alias is not sustainable option for a open-source project documentation. You can't just use `pn` in documentation without confusion

**Reply by zkochan** [reply]
I think I have asked this in the past but I have created a new poll: https://x.com/pnpmjs/status/1953456956095570428

**Reply by brianjlacy** [reply]
I've deleted my Twitter acct in the past year but you can count a +1-in-spirit for the Yes vote on your poll. 😁

**Reply by zkochan** [reply]
43% voted to ship "pn". So, I guess we should? Those 57% can just continue using "pnpm", I guess.

**Reply by roll** [reply]
Just for context, in Python it seems to be a leading solution for package management now is: `uv` and `uvx` (`npx` counterpart)

**Reply by roll** [reply]
Hi, any new on this one?

**Reply by zkochan** [reply]
There was no big excitement about it during the poll. So, I am not confident that we need this

**Reply by roll** [reply]
Now, there is also Vite Plus with nice commands `vp` and `vpx` (exactly like `uv` and `uvx`) but I'm still leaning towards using `pnpm` as a facade command runner for my projects and teams because it's usually guaranteed to be available everywhere (e.g. on Cloudfalre/etc) and you can create non-opinionated set of commands e.g. `test` for us usually means run all the checks and test not only `vitest`. So we continue to run vite-plus by pnpm.   

So hopefully, at some point we still get something like `pn/pnx`. At the moment, my experience is that I stopped using `alias pn=pnpm` because it leaks everywhere =) If I have this alias, I automatically use it in documentation, CI/CD, colleagues' laptops (which doesn't work of course) so I had to change it to `alias p=pnpm` making it more obvious that it is not a real command.

**Reply by zkochan** [reply]
https://github.com/pnpm/pnpm/pull/11052

**Comment by zkochan** [comment]
will be solved in pnpm v11 with pn and pnx via https://github.com/pnpm/pnpm/pull/11052

**Reply by desaiuditd** [reply]
@zkochan The aliases are not exposed when `pnpm` is installed via Corepack. Is that a bug or something wrong on my end?

**Reply by roll** [reply]
https://github.com/nodejs/corepack/issues/833
